gpii/node_modules/flowManager/src/BrowserChannel.js

```diff
@@ -0,0 +1,37 @@
+(function () {
+
+    "use strict";
+
+    var fluid = require("infusion");
+    var gpii = fluid.registerNamespace("gpii");
+
+    fluid.registerNamespace("gpii.request.flowManager");
+
+    fluid.defaults("gpii.flowManager.browserChannel", {
+        gradeNames: ["fluid.standardRelayComponent", "autoInit"]
+    });
+
+    fluid.defaults("kettle.requests.request.handler.browserChannel", {
+        gradeNames: ["gpii.request.flowManager.sessionAware", "autoInit"],
+        invokers: {
+            handle: {
+                funcName: "gpii.request.flowManager.browserChannelHandle",
+                args: ["{request}"],
+                dynamic: true
+            }
+        }
+    });
+
+    gpii.request.flowManager.browserChannelHandle = function (fluidRequest) {
+        var data = fluidRequest.data;
```

Comment:
Should make some attempt to validate the data here. It should correspond to something in the solutions registry or at least be a string of the appropriate form - otherwise we are subject to obvious DoS attacks